src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,156 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            head = new Node<>(null, value, null);
+            tail = head;
+        } else {
+            tail.next = new Node<>(tail, value, null);
+            tail = tail.next;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index " + index + " is out of bound for length "
+                    + size);
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index == 0) {
+            Node<T> newNode = new Node<>(head.prev, value, head);
+            head.prev = newNode;
+            head = newNode;
+            size++;
+            return;
+        }
+        Node<T> currentNode = getCurrentNode(index);
+        Node<T> anotherNode = new Node<>(currentNode.prev, value, currentNode);
+        currentNode.prev.next = anotherNode;
+        currentNode.prev = anotherNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list == null) {
+            throw new RuntimeException("Current list is null!");
+        }
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> currentNode = getCurrentNode(index);
+        return currentNode.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> currentNode = getCurrentNode(index);
+        T oldValue = currentNode.value;
+        currentNode.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> currentNode = getCurrentNode(index);
+        return unlink(currentNode);
     }
 
     @Override
     public boolean remove(T object) {
+        if (size == 1) {
+            remove(size - 1);
+            return true;
+        }
```

Comment:
This could lead to a potential bug. Think of what will happen, when your list contains only 1 element, which does not equal a passed argument.